src/core/sys/posix/sys/time.d

```diff
@@ -28,7 +28,55 @@ else version (WatchOS)
     version = Darwin;
 
 version (Posix):
-extern (C) nothrow @nogc:
+nothrow @nogc:
+
+
+/* macros provided to operate on timeval structures
+ * they are extern(D) because they are not really C symbols, just macros
+ */
+
+pure @safe void timeradd(const timeval* a, const timeval* b,
+        timeval* result)
+{
+    result.tv_sec = a.tv_sec + b.tv_sec;
+    result.tv_usec = a.tv_usec + b.tv_usec;
+    if (result.tv_usec >= 1000000)
+    {
+        ++result.tv_sec;
+        result.tv_usec -= 1000000;
+    }
+}
+
+pure @safe void timersub(const timeval* a, const timeval* b,
+        timeval *result)
+{
+    result.tv_sec = a.tv_sec - b.tv_sec;
+    result.tv_usec = a.tv_usec - b.tv_usec;
+    if (result.tv_usec < 0) {
+        --result.tv_sec;
+        result.tv_usec += 1000000;
+    }
+}
+
+pure @safe void timerclear(timeval* tvp)
+{
+    (tvp.tv_sec = tvp.tv_usec = 0);
+}
+
+pure @safe int timerisset(timeval* tvp)
+{
+    return cast(int) (tvp.tv_sec || tvp.tv_usec);
+}
+
+pure @safe int timercmp(char[] CMP)(const timeval* a, const timeval* b)
+{
+    return cast(int)
+           mixin("((a.tv_sec == b.tv_sec) ?" ~
+                 "(a.tv_usec" ~ CMP ~ "b.tv_usec) :" ~
+                 "(a.tv_sec"  ~ CMP ~ "b.tv_sec))");
+}
```

Comment:
Again, this is taken almost literally from C macros. Is the closest D translation I could think of, I'm not particularly proud of it :P